### PR TITLE
[3006.x] Add new maintenance policy

### PR DESCRIPTION
**Pull request by barbaricyawps**
### What does this PR do?
This PR adds the new proposed maintenance policies for Salt Project.

### What issues does this PR fix or reference?
Fixes: N/A

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [X] Docs
- [X] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html - N/A
- [X] Tests written/updated - N/A

### Commits signed with GPG?
Yes

Please review [Salt's Contributing Guide](https://docs.saltproject.io/en/master/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

File: POLICY.rst [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+We have decided to update our issue policy to be similar to RedHat community
+project policies.
+
+Community members who repeatedly violate these policies are subject to bans.
+
+#. All issues that were not opened against a currently supported release of Salt
+   will be closed.
+
+   - When an old release of Salt is marked out of support, all issues opened
+     against the the now defunct release will be closed.
```
Comment (s0undt3ch) [inline POLICY.rst:30]:
```suggestion
     against the now defunct release will be closed.

```

**Review by s0undt3ch — CHANGES_REQUESTED** [review]

File: POLICY.rst (hunk 1 above)
Comment (barbaricyawps) [reply POLICY.rst]: Thanks for the catch! I made this change on my own branch and pushed it up before I realized this was a commit suggestion. I tend to prefer to work in my own branch anyhow. It should be resolved now!

**Comment by s0undt3ch** [comment]
The pre-commit failures are valid, https://github.com/saltstack/salt/actions/runs/7558398155/job/20580953190?pr=65881#step:10:11

**Comment by barbaricyawps** [comment]
Oh, right. I forgot to run pre-commit. Let me do that.

**Comment by barbaricyawps** [comment]
I just pushed updates to fix pre-commit.

File: POLICY.rst [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+     by the issue submitter and assigned to a release. Otherwise they will be
+     immediately closed.
+   - Discussions about features can be held in the GitHub
+     `Discussions <https://github.com/saltstack/salt/discussions>`_ tab or in
+     the community `Open Hour <https://saltproject.io/calendar/>`_.
+   - Questions will be immediately closed.
+#. Issues must submit sufficient information.
+
+   - Issues must follow the relevant template for information.
+   - Issues that do not give sufficient information about the nature of the
```
Comment (dmurphy18) [inline POLICY.rst]: Wondering if this should state as follows:

Issues that do not give sufficient information about the nature of the issue **and how to reproduce the issue** will be immediately closed.

Change in **BOLD**

Comment (barbaricyawps) [reply POLICY.rst]: I can add that. Sorry that this can't be merged any more. I'll add it manually.

**Review by dmurphy18 — DISMISSED** [review]

**Review by twangboy — DISMISSED** [review]

**Review by twangboy — APPROVED** [review]

**Pull request merged**